src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfTTypeToConvert.cs

```diff
@@ -124,7 +124,6 @@ public override bool GetMemberAndWriteJsonExtensionData(object obj, ref WriteSta
             }
             else
             {
-                state.Current.PolymorphicJsonPropertyInfo = state.Current.DeclaredJsonPropertyInfo!.RuntimeClassInfo.ElementClassInfo!.PolicyProperty;
```

Comment:
@steveharter This was causing me problems since it causes that the `JsonClassInfo` takes its value from `PolymorphicJsonPropertyInfo` when processing the `JsonExtensionData` dictionary, I removed the line and ran the tests and nothing broke, so maybe you might know better what was the purpose behind this.

Reply:
cc @layomia